pkg/plugins/gateway/algorithms/pd_disaggregation.go

```diff
@@ -93,109 +96,151 @@ func (r pdRouter) Route(ctx *types.RoutingContext, readyPodList types.PodList) (
 	decodePod, _ := utils.SelectRandomPod(decodePods, rand.Intn)
 	return decodePod
 }
 
-func (r *pdRouter) doPrefillRequest(routingCtx *types.RoutingContext, prefillPods []*v1.Pod) error {
-	// TODO: refactor constants post PR: 1293 merge
-	if getLLMEngine(prefillPods[0], LLMEngineLabel, "vllm") == "xllm" {
-		return nil
-	}
-
-	var prefillPod *v1.Pod
-	prefixMatchPrefillPods, prefixHashes, err := r.evaluatePrefixCache(routingCtx, prefillPods)
+func (r *pdRouter) doPrefillRequest(routingCtx *types.RoutingContext, prefillPods []*v1.Pod, llmEngine string) error {
+	prefillPod, prefixHashes, err := r.evaluatePrefixCache(routingCtx, prefillPods)
 	if err != nil {
 		return err
 	}
-
-	if len(prefixMatchPrefillPods) > 0 {
-		prefillPod = getTargetPodFromMatchedPods(r.cache, prefillPods, prefixMatchPrefillPods)
-	}
-	if prefillPod == nil {
-		prefillPod = selectTargetPodWithLeastRequestCount(r.cache, prefillPods)
-	}
-
 	defer func() {
 		if len(prefixHashes) > 0 {
 			r.prefixCacheIndexer.AddPrefix(prefixHashes, routingCtx.Model, prefillPod.Name)
 		}
 	}()
 
-	klog.InfoS("doPrefillRequest", "prefill_pod_ip", prefillPod.Status.PodIP)
+	// Prepare prefill request payload
+	payload, err := r.preparePrefillPayload(routingCtx, prefillPod, llmEngine)
+	if err != nil {
+		return fmt.Errorf("failed to prepare prefill payload: %w", err)
+	}
+
+	// Execute HTTP request
+	apiURL := fmt.Sprintf("http://%s:%d%s",
+		prefillPod.Status.PodIP,
+		utils.GetModelPortForPod(routingCtx.RequestID, prefillPod),
+		routingCtx.ReqPath)
+
+	klog.InfoS("start_prefill_request",
+		"request_id", routingCtx.RequestID,
+		"llm_engine", llmEngine,
+		"prefill_url", apiURL)
+
+	if llmEngine == SGLangEngine {
+		go func() {
+			if err := r.executeHTTPRequest(apiURL, routingCtx, payload); err != nil {
+				klog.ErrorS(err, "prefill request for sglang failed", "request_id", routingCtx.RequestID)
+			}
+			klog.InfoS("prefill_request_complete", "request_id", routingCtx.RequestID)
+		}()
+	} else {
+		if err := r.executeHTTPRequest(apiURL, routingCtx, payload); err != nil {
+			return fmt.Errorf("failed to execute prefill request: %w", err)
+		}
+		klog.InfoS("prefill_request_complete", "request_id", routingCtx.RequestID)
+	}
 
-	// Prepare request payload
+	return nil
+}
+
+func (r *pdRouter) preparePrefillPayload(routingCtx *types.RoutingContext, pod *v1.Pod, llmEngine string) ([]byte, error) {
 	var completionRequest map[string]any
 	if err := json.Unmarshal(routingCtx.ReqBody, &completionRequest); err != nil {
-		return fmt.Errorf("failed to unmarshal request body for prefill: %w", err)
+		return nil, fmt.Errorf("failed to unmarshal prefill request body: %w", err)
+	}
+
+	// Handle SGLang specific configuration
+	if llmEngine == SGLangEngine {
+		completionRequest["bootstrap_host"] = pod.Status.PodIP
+		completionRequest["bootstrap_port"] = getSGLangBootstrapPort(pod)
+		completionRequest["bootstrap_room"] = rand.Int63n(1<<63 - 1)
```

Comment:
could you double check sglang router logics? I remember this is used to identify the kv on prefill server.